run_dev_backend.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+script_dir=$(dirname $0)
+backend_pid=""
+
+if [ "$1" == "cypress" ] && [ "z$2" == "z" ]; then
```

Comment:
prefixing with something, and prefixing with z, is quite common in bash, albeit not so useful here with the quotes

Reply:
it's useful without the quotes because if I remember correctly [ $hello = hello ] would be a syntax error if $hello is empty, while [ z$hello = zhello ] wouldn't.
But honestly just use the quotes everywhere in shells, non quoted things are timed bombs

Reply:
Actually, it also guards (guarded? Since this does not seem to apply anymore) against the possibility of the variable containing something that looks like an option.

for instance "$1" could contain "-z".

It seems to be a thing of the past and test / test builtins seems to handle `[ -z = -z ]` correctly.

Some insight: https://stackoverflow.com/questions/6852612/test-for-empty-string-with-x